ddspipe_core/src/cpp/core/DdsPipe.cpp

```diff
@@ -347,18 +345,41 @@ void DdsPipe::updated_endpoint_nts_(
     }
 }
 
+bool DdsPipe::is_endpoint_kind_relevant_(
+        const Endpoint& endpoint) noexcept
+{
+    switch (configuration_.discovery_trigger)
+    {
+        case DiscoveryTrigger::READER:
+            return endpoint.is_reader();
+
+        case DiscoveryTrigger::WRITER:
+            return endpoint.is_writer();
+
+        case DiscoveryTrigger::ANY:
+            return true;
+
+        case DiscoveryTrigger::NONE:
+            return false;
+
+        default:
+            utils::tsnh(utils::Formatter() << "Invalid Discovery Trigger.");
+            return false;
+    }
+}
+
 bool DdsPipe::is_endpoint_relevant_(
         const Endpoint& endpoint) noexcept
 {
-    if (!endpoint.is_reader())
+    if (!is_endpoint_kind_relevant_(endpoint))
     {
         return false;
     }
 
-    auto is_endpoint_relevant = [endpoint](const Endpoint& entity)
+    auto is_endpoint_relevant = [&, endpoint](const Endpoint& entity)
             {
                 return entity.active &&
-                       entity.is_reader() &&
+                       is_endpoint_kind_relevant_(entity) &&
```

Comment:
This makes little sense to me, as this logic specific to the remove unused entities feature is not supported for a discovery trigger value other than `READER`. But ok, it will need to be this way if we ever add support. However, I think it would be good to fail (add condition to some `is_valid`) if an unsupported configuration is provided.

Reply:
Related suggestion, although does not tackle the previous comment (cannot hold it back): I suggest having an if-else with `remove_unused_entities` within `is_relevant` method, and return `endpoint.active` when false.

Reply:
I think it is more appropriate to address that change in a new branch, since it changes the dynamic of the DdsPipe at it should be thoroughly thought about and tested. I will also take the liberty to change other things which I think would improve the DdsPipe readability.

Reply:
Write an error trace in is_valid

Reply:
"Different to" is apparently colloquial, use "from" or "than".
https://dictionary.cambridge.org/es/gramatica/gramatica-britanica/different-from-different-to-or-different-than
https://www.wordreference.com/es/translation.asp?tranword=different%20to